### PR TITLE
Adds Sentry Performance Monitoring Capabilities

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR adds Sentry Performance Monitoring capabilities that's managed through a Firebase remote config. It's disabled for debug builds as well as for users who has tracking disabled. It's also disabled by default and will only work if we set the `wp_android_performance_monitoring_sample_rate` remote config value in the range of `0 < sampleRate <=1.0`. The implementation closely follows it's counterpart from `WooCommerce-Android` project.

The remote config value is set to `0.0` for now. I intend to keep it this way and let folks who are interested in this capability to change the sampling rate as they see fit. This is to prevent us from accidentally sending too many events and getting a large bill to pay.

Note that the `TransactionLauncher`s and transactions for HTTP requests are out of scope for this PR. We'll need to have an internal discussion about that before we add them.

**To test:**

Admittedly I don't know how we should test this implementation. The implementation is unit tested and following the implementation from WCAndroid, so I am hopeful that it will work. However, if there is a good way to test this, I'd like to do that before we merge the PR.

@wzieba I'd love to hear your thoughts on testing since you worked on this for WCAndroid. Thanks in advance!

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

**UI Changes testing checklist:**

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [x] Languages with large words or with letters/accents not frequently used in English.
- [x] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [x] Large and small screen sizes. (Tablet and smaller phones)
- [x] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/util/crashlogging/WPPerformanceMonitoringConfigTest.kt
Comment (ParaskP7) [inline WordPress/src/test/java/org/wordpress/android/util/crashlogging/WPPerformanceMonitoringConfigTest.kt:15]: Praise (❤️): Awesome work on the tests, many thanks for covering as many scenarios as possible! 🥇

File: WordPress/src/main/java/org/wordpress/android/util/FirebaseRemoteConfigWrapper.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/util/FirebaseRemoteConfigWrapper.kt:17]: Minor (🔍): Not related to your change, but maybe we should consider making this constant `private` as well. Feel free to totally ignore this comment. 😊

**Review by ParaskP7 — APPROVED** [review]
👋 @oguzkocer !

I have reviewed this PR as per the instructions, everything LGTM and in par with WCAndroid. Good job bringing Sentry's performance monitoring to WPAndroid and making it configurable via Firebase remote config! 🌟 💯 🥇 

-----

> Admittedly I don't know how we should test this implementation. The implementation is unit tested and following the implementation from WCAndroid, so I am hopeful that it will work. However, if there is a good way to test this, I'd like to do that before we merge the PR.

Same! 😊 

PS: All I did is verifying that the Firebase related `wp_android_performance_monitoring_sample_rate` remote config has been indeed created and has the expected parameter values.

> @wzieba I'd love to hear your thoughts on testing since you worked on this for WCAndroid. Thanks in advance!

This! ⬆️ 

PS: I am going to approve this PR, just to unblock it, but getting @wzieba input before merging is very welcome.

**Comment by ParaskP7** [comment]
😞 Ah, this just got merged due to auto-merged being enabled. I usually don't check for auto-merge being enabled on such PRs, especially when extra feedback is waited upon, apologies @oguzkocer , I should have checked anyway! 😭

**Comment by wzieba** [comment]
👋 I don't see anything blocking in this PR so I think the merge was safe!

What I could suggest is:

1. Integrate `CrashLoggingOkHttpInterceptor` to have HTTP spans added in performance transaction. I believe that for our mobile apps, it's the most crucial part.

2. You can enable adding database operations spans by updating Sentry Gradle Plugin settings:

```groovy
sentry {
    tracingInstrumentation {
        enabled = true
        features = [io.sentry.android.gradle.extensions.InstrumentationFeature.DATABASE]
    }
}
```


About testing: Sentry adds default transactions for all activities and start of the app. I'm installing the app now to check if the integration works.

**Comment by wzieba** [comment]
It's alive!

image

**Comment by wzieba** [comment]
I also see that the network interceptor was added automatically by Sentry.

It's good but not perfect - adding our own network interceptor allows us to specify our own `com.automattic.android.tracks.crashlogging.RequestFormatter`. We should use `RequestFormatter` to remove data like `blog_id` or `post_id` from requests to:
- increase privacy
- make it possible for Sentry to merge the same types of network spans when it analyzes performance transactions.

A sample `RequestFormatter` can be found [here](https://github.com/woocommerce/woocommerce-android/blob/trunk/WooCommerce/src/main/kotlin/com/woocommerce/android/performance/WooRequestFormatter.kt) - see unit tests to see how it works. I _think_ it can be safely reused in WordPress / Jetpack apps. 

--- 

I see that Sentry also enables, by default, database operations spans, but I think they're safe to leave as it is.

File: WordPress/src/test/java/org/wordpress/android/util/crashlogging/WPPerformanceMonitoringConfigTest.kt
Comment (oguzkocer) [reply WordPress/src/test/java/org/wordpress/android/util/crashlogging/WPPerformanceMonitoringConfigTest.kt:15]: I borrowed 3/4 of these tests from @wzieba's implementation ❤️

**Comment by oguzkocer** [comment]
@ParaskP7 @wzieba Thank you both for the review! I shouldn't have enabled auto-merge for this PR, sorry about that!

I'm working on a follow up PR to apply the suggestions.

**Pull request merged**
